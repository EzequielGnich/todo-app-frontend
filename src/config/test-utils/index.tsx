import { render, RenderOptions } from "@testing-library/react";
import { ConfigProvider } from "antd";
import React, { ReactElement } from "react";
import { ReactQueryProvider } from "../react-query";

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return (
    <ReactQueryProvider>
      <ConfigProvider>{children}</ConfigProvider>
    </ReactQueryProvider>
  );
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
) => render(ui, { wrapper: AllTheProviders, ...options });

export * from "@testing-library/react";
export { customRender as render };
