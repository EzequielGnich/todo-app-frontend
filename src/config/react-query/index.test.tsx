import { ReactQueryProvider } from ".";
import { render } from "../test-utils";

const renderComponent = () => {
  return render(
    <ReactQueryProvider>
      <div>Test</div>
    </ReactQueryProvider>
  );
};

describe("React query provider", () => {
  it("should render", () => {
    const { container } = renderComponent();
    expect(container).toMatchSnapshot();
  });
});
