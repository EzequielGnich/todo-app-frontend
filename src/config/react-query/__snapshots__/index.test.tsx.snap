// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`React query provider > should render 1`] = `
<div>
  <div>
    Test
  </div>
</div>
`;
